cmd2.py

```diff
@@ -438,6 +438,229 @@ def strip_ansi(text):
+                    submenu.postcmd(ret, line)
+                else:
+                    history = _pop_readline_history()
+                    if self.reformat_prompt is not None:
+                        prompt = submenu.prompt
+                        submenu.prompt = self.reformat_prompt.format(
+                            super_prompt=parent_cmd.prompt,
+                            command=self.command,
+                            sub_prompt=prompt,
+                        )
+                    submenu.cmdloop()
+                    if self.reformat_prompt is not None:
+                        self.submenu.prompt = prompt
+                    _push_readline_history(history)
+            finally:
+                # copy back original attributes
+                for attr, value in original_attributes.items():
+                    if attr is not AddSubmenu._Nonexistent:
+                        setattr(submenu, attr, value)
+                    else:
+                        delattr(submenu, attr)
+
+        def complete_submenu(_self, text, line, begidx, endidx):
+            """
+            This function will be bound to complete_<submenu> and will perform the complete commands of the submenu.
+            """
+            submenu = self.submenu
+            original_attributes = {
+                attr: getattr(submenu, attr, AddSubmenu._Nonexistent)
+                for attr in self.shared_attributes.keys()
+            }
+            try:
+                # copy over any shared attributes
+                for sub_attr, par_attr in self.shared_attributes.items():
+                    setattr(submenu, sub_attr, getattr(_self, par_attr))
+
+                return _complete_from_cmd(submenu, text, line, begidx, endidx)
+            finally:
+                # copy back original attributes
+                for attr, value in original_attributes.items():
+                    if attr is not AddSubmenu._Nonexistent:
+                        setattr(submenu, attr, value)
+                    else:
+                        delattr(submenu, attr)
+
+        original_do_help = Cmd.do_help
+        original_complete_help = Cmd.complete_help
+
+        def help_submenu(_self, line):
+            """
+            This function will be bound to help_<submenu> and will call the help commands of the submenu.
+            """
+            tokens = line.split(None, 1)
+            if tokens and (tokens[0] == self.command or tokens[0] in self.aliases):
+                self.submenu.do_help(tokens[1] if len(tokens) == 2 else '')
+            else:
+                original_do_help(_self, line)
+
+        def _complete_submenu_help(_self, text, line, begidx, endidx):
+            """autocomplete to match help_submenu()'s behavior"""
+            tokens = line.split(None, 1)
+            if len(tokens) == 2 and (
+                tokens[1].startswith(self.command) or
+                any(tokens[1].startswith(alias) for alias in self.aliases)
```

Comment:
alias local variable name shadows name from outer scope of __call__ implementation below, best to be clear and rename one of them